src/workerd/io/worker-modules.h

```diff
@@ -371,6 +371,67 @@ kj::Maybe<jsg::ModuleRegistry::ModuleInfo> tryCompileLegacyModule(jsg::Lock& js,
   KJ_UNREACHABLE;
 }
 
+template <typename JsgIsolate>
+kj::Array<Worker::Script::CompiledGlobal> compileServiceWorkerGlobals(jsg::Lock& js,
+    const Worker::Script::ScriptSource& source,
+    const Worker::Isolate& isolate,
+    const jsg::CompilationObserver& observer) {
+  auto& lock = kj::downcast<typename JsgIsolate::Lock>(js);
+
+  auto globals = source.globals.asPtr();
+  auto compiledGlobals = kj::heapArrayBuilder<Worker::Script::CompiledGlobal>(globals.size());
+
+  for (auto& global: globals) {
+    js.withinHandleScope([&] {
+      // Don't use String's usual TypeHandler here because we want to intern the string.
+      auto name = jsg::v8StrIntern(js.v8Isolate, global.name);
+
+      v8::Local<v8::Value> value;
+
+      KJ_SWITCH_ONEOF(global.content) {
+        KJ_CASE_ONEOF(content, Worker::Script::TextModule) {
+          value =
+              workerd::modules::legacy::template compileTextGlobal<JsgIsolate>(lock, content.body);
+        }
+        KJ_CASE_ONEOF(content, Worker::Script::DataModule) {
+          value =
+              workerd::modules::legacy::template compileDataGlobal<JsgIsolate>(lock, content.body);
+        }
+        KJ_CASE_ONEOF(content, Worker::Script::WasmModule) {
+          value = workerd::modules::legacy::template compileWasmGlobal<JsgIsolate>(
+              lock, content.body, observer);
+        }
+        KJ_CASE_ONEOF(content, Worker::Script::JsonModule) {
+          value =
+              workerd::modules::legacy::template compileJsonGlobal<JsgIsolate>(lock, content.body);
+        }
```

Comment:
Am I understanding this right that this enables more global types for scripts? Can we have tests for these?

Reply:
Ah I see it's coming from Edgeworker, nvm.